Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,930 @@
+* @param np2 an optional sequence of \ref bgl_namedparameters "Named Parameters 2" of the `target` providing a vertex point map and a vertex normal map as listed below.
+*
+* \cgalNamedParamsBegin{Named Parameters 1}
+*   \cgalParamNBegin{number_of_iterations}
+*     \cgalParamDescription{the number of registration iterations using ICP, iterative closest point}
+*     \cgalParamType{unsigned int}
+*     \cgalParamDefault{`50`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{point_to_plane_energy}
+*     \cgalParamDescription{the weight of the point to plane distance in the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{point_to_point_energy}
+*     \cgalParamDescription{the weight of the point to matching point distance in the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{as_rigid_as_possible_energy}
+*     \cgalParamDescription{defines the rigidity of the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`50`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{maximum_matching_distance}
+*     \cgalParamDescription{the maximum distance for a vertex in `source` to match with a point in `target`. The default value 0 means no maximum matching distance.}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`0`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{correspondences}
+*     \cgalParamDescription{a range of matching vertex pairs between the `source` and the `target`.}
+*     \cgalParamType{`ConstRange` whose value type is a pair of `boost::graph_traits<TriangleMesh1>::%vertex_descriptor` and `boost::graph_traits<TriangleMesh2>::%vertex_descriptor`.}
+*     \cgalParamDefault{empty}
+*     \cgalParamExtra{to avoid copies, this parameter can be passed using `std::cref`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_point_map}
+*     \cgalParamDescription{a property map associating points to the vertices of `source`}
+*     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+*                    as key type and `%Point_3` as value type}
+*     \cgalParamDefault{`get_const_property_map(CGAL::vertex_point, source)`}
+*     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+*                     must be available in `TriangleMesh1`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{geom_traits}
+*     \cgalParamDescription{an instance of a geometric traits class}
+*     \cgalParamType{a class model of `Kernel`}
+*     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+*     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+*   \cgalParamNEnd
+* \cgalNamedParamsEnd
+*
+* \cgalNamedParamsBegin{Named Parameters 2}
+*   \cgalParamNBegin{vertex_normal_map}
```

Comment:
check the doc [here](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__corrected__curvatures__grp.html#ga22665c9ce92aaedab07df1b05f20bdb2) if you want to init the vpm

Reply:
ping @soesau